### PR TITLE
wallet: wait for chain sync on Neutrino recovery

### PR DESCRIPTION
**Pull request by guggero**
Normally the wallet doesn't wait for the chain backend to be synced
on regtest/simnet because there we cannot be certain if we are at
the chain tip, as there are no other nodes to compare to. For
Neutrino, this is a bit different because we rely on the cfheader
server to tell us what it thinks the chain tip is. For a wallet
recovery on Neutrino we therefore need to make sure we are at least
synced up to what the server thinks is the tip.

### CODE REVIEW

File: wallet/wallet.go [hunk 1]
```diff
@@ -653,6 +664,9 @@ func (w *Wallet) recovery(chainClient chain.Interface,
 	if err != nil {
 		return err
 	}
+	log.Debugf("Chain backend has best block %d which is current: %v",
+		bestHeight, chainClient.IsCurrent())
+	chainClient.NotifyBlocks()
```
Comment (halseth) [inline wallet/wallet.go]: why is this change needed?

File: wallet/wallet.go [hunk 2]
```diff
@@ -336,10 +336,21 @@ func (w *Wallet) syncWithChain(birthdayStamp *waddrmgr.BlockStamp) error {
...
+		w.recoveryWindow > 0
+
 	// We'll wait until the backend is synced to ensure we get the latest
 	// MaxReorgDepth blocks to store. We don't do this for development
-	// environments as we can't guarantee a lively chain.
-	if !w.isDevEnv() {
+	// environments as we can't guarantee a lively chain, except for
+	// Neutrino, where the cfheader server tells us what it believes the
+	// chain tip is.
+	if !w.isDevEnv() || neutrinoRecovery {
```
Comment (halseth) [inline wallet/wallet.go]: Should consider removing this special casing at one point, but for now this is an okay fix I think.

File: wallet/wallet.go (hunk 1 above)
Comment (guggero) [reply wallet/wallet.go]: Oops, that was some debugging/trial-and-error code I committed by accident. Removed.
